src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsUapHiViMobileWiki.java

```diff
@@ -0,0 +1,274 @@
+    TimeUnit.SECONDS.sleep(1);
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+
+    adsPage.scrollBy(0, -500);
+    TimeUnit.SECONDS.sleep(1);
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldAutoplayVideoForImpactState() {
+    openPage();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.waitForVideoStart();
+  }
+
+  @Test
+  public void shouldProgressTimeWhilePlaying() throws InterruptedException {
+    openPage();
+
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.waitForVideoStart();
+
+    hiViUap.clickVideo();
+    final double startProgressBarWidth = hiViUap.getProgressBarWidth();
+    TimeUnit.SECONDS.sleep(3);
+    hiViUap.clickVideo();
+
+    Assert.assertTrue(startProgressBarWidth < hiViUap.getProgressBarWidth(), "Video time indicator should move.");
+  }
+
+  @Test
+  public void shouldMuteVideoForAutoplayedImpactState() throws InterruptedException {
+    openPage();
+    new HiViUap(driver, TLB_SLOT_ID);
+
+    TimeUnit.SECONDS.sleep(3);
+    Assertion.assertFalse(SoundMonitor.wasSoundHeardOnPage(new JavascriptActions()));
+    TimeUnit.SECONDS.sleep(3);
+    Assertion.assertFalse(SoundMonitor.wasSoundHeardOnPage(new JavascriptActions()));
+  }
+
+  @Test
+  public void shouldUnmuteVideoAfterClickOnIcon() throws InterruptedException {
+    openPage();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+
+    hiViUap.waitForVideoStart();
+    hiViUap.clickVideo();
+    hiViUap.toggleSound();
+    TimeUnit.SECONDS.sleep(3);
+
+    Assertion.assertTrue(SoundMonitor.wasSoundHeardOnPage(new JavascriptActions()));
+  }
+
+  @Test
+  public void shouldPlayUnmutedVideoForReplayedAd() throws InterruptedException {
+    openPage();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.waitForVideoStart();
+    hiViUap.waitForVideoEnd();
+    hiViUap.clickReplayButton();
+    TimeUnit.SECONDS.sleep(3);
+
+    Assertion.assertTrue(SoundMonitor.wasSoundHeardOnPage(new JavascriptActions()));
+  }
+
+  @Test
+  public void shouldRedirectToPageAfterClickOnAd() {
+    AdsBaseObject adsPage = openPageOnSecondPV();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.clickVideo();
+    hiViUap.clickAd();
+
+    Assert.assertTrue(adsPage.tabContainsUrl(AD_REDIRECT));
+  }
+
+  @Test
+  public void shouldRedirectAfterClickOnLearnMore() {
+    AdsBaseObject adsPage = new AdsBaseObject(driver, AdsDataProvider.UAP_HIVI_PAGE.getUrl());
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+    hiViUap.clickLearnMore();
+
+    Assert.assertTrue(adsPage.tabContainsUrl(AD_REDIRECT));
+  }
+
+  @Test
+  public void shouldPauseOnVideoAfterClickOnPauseIcon() throws InterruptedException {
+    openPage();
+    HiViUap hiViUap = new HiViUap(driver, TLB_SLOT_ID);
+
+    hiViUap.waitForVideoStart();
+    TimeUnit.SECONDS.sleep(2);
+    hiViUap.clickVideo(); // to trigger ui
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)